test/functional/test_framework/wallet.py

```diff
@@ -414,11 +416,13 @@ def getnewdestination(address_type='bech32m'):
 
 def address_to_scriptpubkey(address):
     """Converts a given address to the corresponding output script (scriptPubKey)."""
+    if address[:4] == 'tb1q': # testnet segwit
```

Comment:
thinking about this a bit more, I think we should have a more robust way of checking this. iirc, taproot would be `tb1p` on testnet, so this would not work for segwit v1 addresses.

Maybe instead you could do something like `version, payload = bech32_to_bytes` and then check the version. If the version is None, then move on to `payload, version = base58_to_byte` etc.

Reply:
Thanks for the thorough review Josi.
I will fix that.
This means I have to check in `bech32_to_bytes `also for a None version value after   so as not to call bytearray(payload) with None.